apisix/plugins/loggly.lua

```diff
@@ -0,0 +1,243 @@
+    ALERT = 1,              --  action must be taken immediately
+    CRIT = 2,               --  critical conditions
+    ERR = 3,                --  error conditions
+    WARNING = 4,            --  warning conditions
+    NOTICE = 5,             --  normal but significant condition
+    INFO = 6,               --  informational
+    DEBUG = 7,              --  debug-level messages
+}
+
+local schema = {
+    type = "object",
+    properties = {
+        customer_token = {type = "string"},
+        severity = {
+            type = "string",
+            default = "INFO",
+            enum = {"DEBUG", "INFO", "NOTICE", "WARNING", "ERR", "CRIT", "ALERT", "EMEGR",
+                    "debug", "info", "notice", "warning", "err", "crit", "alert", "emegr"},
+            description = "base severity log level",
+        },
+        include_req_body = {type = "boolean", default = false},
+        include_resp_body = {type = "boolean", default = false},
+        include_resp_body_expr = {
+            type = "array",
+            minItems = 1,
+            items = {
+                type = "array"
+            }
+        },
+        tags = {
+            type = "array",
+            minItems = 1,
+            items = {
+                type = "string"
+            }
+        },
+        prefer_name = {type = "boolean", default = true}
+    },
+    required = {"customer_token"}
+}
+
+
+local defaults = {
+    host = "logs-01.loggly.com",
+    port = 514,
+    protocol = "syslog",
+    timeout = 5000
+}
+
+local metadata_schema = {
+    type = "object",
+    properties = {
+        host = {
+            type = "string",
+            default = defaults.host
+        },
+        port = {
+            type = "integer",
+            default = defaults.port
+        },
+        protocol = {
+            type = "string",
+            default = defaults.protocol,
+            -- more methods coming soon
+            enum = {"syslog"}
+        },
+        timeout = {
+            type = "integer",
+            minimum = 1,
+            default= defaults.timeout
+        },
+        log_format = {
+            type = "object",
+        }
+    }
+}
+
+local _M = {
+    version = 0.1,
+    priority = 411,
+    name = plugin_name,
+    schema = batch_processor_manager:wrap_schema(schema),
+    metadata_schema = metadata_schema
+}
+
+function _M.check_schema(conf, schema_type)
+    if schema_type == core.schema.TYPE_METADATA then
+        return core.schema.check(metadata_schema, conf)
+    end
+    return core.schema.check(schema, conf)
+end
+
+local function generate_log_message(conf, ctx)
+    local metadata = plugin.plugin_metadata(plugin_name)
+    local entry
+
+    if metadata and metadata.value.log_format
+       and core.table.nkeys(metadata.value.log_format) > 0
+    then
+        entry = log_util.get_custom_format_log(ctx, metadata.value.log_format)
+    else
+        entry = log_util.get_full_log(ngx, conf)
+    end
+
+    if conf.prefer_name then
```

Comment:
It seems we don't need this as we support customizing format?

Reply:
Ack.